internal/api-key/command_delete.go

```diff
@@ -28,17 +26,38 @@ func (c *command) newDeleteCommand() *cobra.Command {
 
 func (c *command) delete(cmd *cobra.Command, args []string) error {
 	c.setKeyStoreIfNil()
-	apiKey := args[0]
 
-	promptMsg := fmt.Sprintf(errors.DeleteResourceConfirmYesNoMsg, resource.ApiKey, apiKey)
-	if ok, err := form.ConfirmDeletion(cmd, promptMsg, ""); err != nil || !ok {
+	existenceFunc := func(id string) bool {
+		_, _, err := c.V2Client.GetApiKey(id)
+		return err == nil
+	}
+
+	if err := deletion.ValidateAndConfirmDeletionYesNo(cmd, args, existenceFunc, resource.ApiKey); err != nil {
 		return err
 	}
-	httpResp, err := c.V2Client.DeleteApiKey(apiKey)
-	if err != nil {
-		return errors.CatchApiKeyForbiddenAccessError(err, deleteOperation, httpResp)
+
+	deleteFunc := func(id string) error {
+		if r, err := c.V2Client.DeleteApiKey(id); err != nil {
```

Comment:
I think we've standardized around this
```suggestion
		if httpResp, err := c.V2Client.DeleteApiKey(id); err != nil {
```